packages/core/src/components/panel-stack2/panelStack2.tsx

```diff
@@ -97,19 +97,16 @@ export const PanelStack2: PanelStack2Component = <T extends Panel<object>>(props
         stack: propsStack,
     } = props;
     const isControlled = propsStack != null;
-    const [direction, setDirection] = React.useState("push");
 
     const [localStack, setLocalStack] = React.useState<T[]>(initialPanel !== undefined ? [initialPanel] : []);
     const stack = React.useMemo(
         () => (isControlled ? propsStack.slice().reverse() : localStack),
         [localStack, isControlled, propsStack],
     );
     const stackLength = React.useRef<number>(stack.length);
```

Comment:
nit: related to below and existing but this is really `prevStackLength`, we could clean this up with something like

```
// default prevStackLength to current length on first render to avoid handling undefined separately
const prevStackLength = usePrevious(stack.length) ?? stack.length;
```